### PR TITLE
Initial geometric measures for fibers

### PR DESCRIPTION
**Pull request by mogres**
This PR adds base functions to calculate geometric measures for fibers denoted by an array of fiber points at a given time.

Utilities added:
1. Function to calculate distance of points from the axis, length of projection along the axis, and position of projected points

Measures added:
1. Average perpendicular distance from the axis
1. peak asymmetric distance: distance of projection of peak from midpoint
2. total fiber twist: sum of consecutive angles between projections from fiber points to the axis

### CODE REVIEW

**Review by kvegesna — APPROVED** [review]
looks good to me!

File: subcell_analysis/compression_analysis.py [hunk 1]
```diff
@@ -19,3 +18,106 @@ def asymmetry(fibers_df):
...
+    position_vectors = fiber_points - fiber_points[0]
+    dot_products = np.dot(position_vectors, axis)
+
+    projections = dot_products / axis_length
+    projection_positions = fiber_points[0] + projections[:, None] * axis / axis_length
+
+    perp_distances = np.linalg.norm(fiber_points - projection_positions, axis=1)
+    scaled_projections = projections / axis_length
+
+    return perp_distances, scaled_projections, projection_positions
```
Comment (blairlyons) [inline subcell_analysis/compression_analysis.py:74]: Can you provide some doc about the `projection_positions`?

File: subcell_analysis/compression_analysis.py [hunk 2]
```diff
@@ -19,3 +18,106 @@ def asymmetry(fibers_df):
         #print(diff[index])
     print(len(diff))
     #write 
+
+
+def get_axis_distances_and_projections(fiber_points):
```
Comment (blairlyons) [inline subcell_analysis/compression_analysis.py]: would be cool to type these too, so this one would be 
`def get_axis_distances_and_projections(fiber_points: np.ndarray) -> Tuple[np.ndarry,np.ndarry,np.ndarry]:`

File: subcell_analysis/compression_analysis.py [hunk 3]
```diff
@@ -19,3 +18,106 @@ def asymmetry(fibers_df):
...
+    """
+    Returns the distances of fiber points from the axis
```
Comment (blairlyons) [inline subcell_analysis/compression_analysis.py]: By axis, you mean the higher order axis the control points are wrapping around, not the fiber axis itself? Would be good to clarify that, maybe nice to also add a few bullets to the readme with vocab, a suggestion:
- Monomer axis: the axis defined by the ReaDDy monomers wrapping the fiber axis
- FIber axis: the fiber axis itself which is defined by the control points
- Higher order axis(?): there's potentially some higher order helix that the fiber itself is wrapping when compressed

**Review by blairlyons — APPROVED** [review]
Very nice! The docstrings are great. They inspired me to ask for a few other nice-to-haves. I'll try to fix the CI in my next PR so we don't have failures. It would be nice to add tests for some of these too, although I know we're in fast R&D mode with this codebase.

File: subcell_analysis/compression_analysis.py (hunk 3 above)
Comment (mogres) [reply subcell_analysis/compression_analysis.py]: Here, I am defining the axis as the line joining the first and last control point. Any suggestions for the nomenclature? Perhaps `end_to_end_axis`? 

The functions in this PR use what we referred to as the "backbone" of the fiber, which are the control points for cytosim, and a derived metric for ReaDDy. I could rename `fiber_points` to `polymer_trace` if that makes more sense.

**Review by mogres — COMMENTED** [review]
Thanks for the suggestions! 
I added some docstrings and definitions to the readme, and formatted everything using `black`. I agree it'd be good to have some tests, and we could look into it either as a group, or I can circle back to it when I get some bandwidth.

File: subcell_analysis/compression_analysis.py (hunk 3 above)
Comment (blairlyons) [reply subcell_analysis/compression_analysis.py]: I think `fiber_points` is a good term for this, I might add a sentence to the readme at some point, but fine for now

**Pull request merged**